### PR TITLE
minor fixes in MLP backprop

### PR DESCRIPTION
**Pull request by navivokaj**
Hi Arthur, nice work on Deep Course! I was going through it to have a refresher on CV and noticed some potential errors on mlp.py:
1.  the softmax in self.fit is redundant since it was already performed in self.forward to produce the probs
2. grad_bh depends on grad_htilde, not grad_h, since tanh is performed on XW+b

### CODE REVIEW

**Comment by arthurdouillard** [comment]
Oops, you're right. Thanks for the catch!
